packages/passport/sdk/src/confirmation/confirmation.ts

```diff
@@ -149,27 +157,84 @@ export default class ConfirmationScreen {
       return;
     }
 
-    this.confirmationWindow = openPopupCenter({
-      url: this.getHref('loading'),
-      title: CONFIRMATION_WINDOW_TITLE,
-      width: popupOptions?.width || CONFIRMATION_WINDOW_WIDTH,
-      height: popupOptions?.height || CONFIRMATION_WINDOW_HEIGHT,
-    });
+    this.popupOptions = popupOptions;
+
+    try {
+      this.confirmationWindow = openPopupCenter({
+        url: this.getHref('loading'),
+        title: CONFIRMATION_WINDOW_TITLE,
+        width: popupOptions?.width || CONFIRMATION_WINDOW_WIDTH,
+        height: popupOptions?.height || CONFIRMATION_WINDOW_HEIGHT,
+      });
+      this.overlay = new Overlay(this.config.popupOverlayOptions);
+    } catch (e) {
+      // If an error is thrown here then the popup is blocked
+      this.overlay = new Overlay(this.config.popupOverlayOptions, true);
+    }
+
+    this.overlay.append(
+      () => {
+        try {
+          this.confirmationWindow?.close();
+          this.confirmationWindow = openPopupCenter({
+            url: this.getHref('loading'),
+            title: CONFIRMATION_WINDOW_TITLE,
+            width: this.popupOptions?.width || CONFIRMATION_WINDOW_WIDTH,
+            height: this.popupOptions?.height || CONFIRMATION_WINDOW_HEIGHT,
+          });
+        } catch { /* Empty */ }
+      },
+      () => this.closeWindow(),
+    );
   }
 
   closeWindow() {
     this.confirmationWindow?.close();
+    this.overlay?.remove();
+    this.overlay = undefined;
   }
 
   showConfirmationScreen(href: string, messageHandler: MessageHandler, resolve: Function) {
-    this.confirmationWindow!.location.href = href;
+    // If popup blocked, the confirmation window will not exist
+    if (this.confirmationWindow) {
+      this.confirmationWindow.location.href = href;
+    }
+
+    // This indicates the user closed the overlay so the transaction should be rejected
+    if (!this.overlay) {
+      resolve({ confirmed: false });
+      return;
+    }
```

Comment:
This doesn't handle the following scenario correctly:
- `loading` is called, but the popup is blocked
- `showConfirmationScreen` is called
- the uses presses the close button on the overlay

In the above scenario, the promise will never resolve